API/authentication/app.js

```diff
@@ -12,159 +13,157 @@ dotenv.config({ path: '.env' });
-                    onSuccess: function (result) {
-                        let accessToken = result.getAccessToken().getJwtToken();
-                        
-                        /* NOTE: Use the idToken for Logins Map when Federating User Pools with identity pools
-                        or when passing through an Authorization Header to an API Gateway Authorizer */
-                        let idToken = result.idToken.jwtToken;
-                        console.log(idToken);
-                        resolve(idToken);
-                    },
-                
-                    onFailure: function(err) {
-                        console.log(err);
-                        reject(err);
-                    }
-                });
-            }
-        );
-    };
-    return cognitoPromise(authenticationDetails);
+  const userPool = new AmazonCognitoIdentity.CognitoUserPool(poolData);
+  const authenticationData = {
+    Username: username,
+    Password: password,
+  };
+  const authenticationDetails = new AmazonCognitoIdentity.AuthenticationDetails(authenticationData);
+  const userData = {
+    Username: username,
+    Pool: userPool,
+  };
+  const cognitoUser = new AmazonCognitoIdentity.CognitoUser(userData);
+
+  const cognitoPromise = (authDetails) => new Promise(
+    (resolve, reject) => {
+      cognitoUser.authenticateUser(authDetails, {
+        onSuccess(result) {
+          /* NOTE: Use the idToken for Logins Map when Federating User Pools with identity pools
+          or when passing through an Authorization Header to an API Gateway Authorizer */
+          console.log(result);
+          resolve(result);
+        },
+
+        onFailure(err) {
+          console.log(err);
+          reject(err);
+        },
+      });
+    },
+  );
+  return cognitoPromise(authenticationDetails);
 }
 
 // Create User and Return User Promise
 async function createUser(username, password, email) {
-    let userPool = new AmazonCognitoIdentity.CognitoUserPool(poolData);
-    let attributeList = [];
-
-    let dataEmail = {
-        Name : 'email',
-        Value : email
-    };
-    
-    let attributeEmail = new AmazonCognitoIdentity.CognitoUserAttribute(dataEmail);
-
-    attributeList.push(attributeEmail);
-
-    // NOTE: userPool.signUp DOES NOT work with promisify - Uses Custom Promise Instead
-    let signupPromise  = (username, password, attributeList) => {
-       return new Promise(
-           (resolve, reject) => {
-               userPool.signUp(username, password, attributeList, null, function (err, result) {
-                   if (err) {
-                       reject(err);
-                   }
-                   else {
-                        resolve(result.user);
-                   }
-               });
-           }
-       )
-    }
-    return signupPromise(username, password, attributeList);
+  const userPool = new AmazonCognitoIdentity.CognitoUserPool(poolData);
+  const attributeList = [];
+
+  const dataEmail = {
```

Comment:
TODO: Comment on what dataEmail is and what atttributeList() is doing.

Reply:
Sure thing, added comments to clarify what those are doing.